File: wechat/src/api/activity.ts
import type {
  WxActivityDetailItem,
  WxActivityListParams,
  WxActivityListResponse,
  WxUserRegisteredActivitiesResponse,
} from '@/api/types/activity'
import type { WxActivityRegistrationResponse } from '@/api/types/activityRegistration'
import { http } from '@/http/http'

// 微信小程序活动API
export function getWxActivityList(params: WxActivityListParams = {}) {
  return http.get<WxActivityListResponse>('/api/wx/Activities', params)
}

// 微信小程序活动详情API
export function getWxActivityDetail(id: number | string) {
  return http.get<WxActivityDetailItem>(`/api/wx/Activities/${id}`)
}

// 活动报名
export function wxRegisterForActivity(activityID: number) {
  return http.post('/api/wx/Activities/register', { activityID })
}

// 创建活动支付订单
export function wxCreateActivityOrder(activityID: number) {
  return http.post<{
    registrationID: number
    appId: string
    timeStamp: string
    nonceStr: string
    package: string
    signType: string
    paySign: string
    amount: number
    orderNo: string
  }>('/api/wx/Activities/create-order', { activityID })
}

// 获取活动报名状态
export function wxGetActivityRegistrationStatus(activityID: number) {
  return http.get<WxActivityRegistrationResponse>('/api/wx/Activities/registration-status', { activityID })
}

// 获取用户已报名活动列表
export function wxGetUserRegisteredActivities(params: {
  page?: number
  pageSize?: number
  paymentStatus?: string
  startTimeRange?: string
} = {}) {
  return http.get<WxUserRegisteredActivitiesResponse>('/api/wx/Activities/registered', params)
}

// 更新参与者信息
export function wxUpdateParticipantInfo(data: {
  registrationID: number
  participantName: string
  participantPhone: string
}) {
  return http.post('/api/wx/Activities/participant-info', data)
}

// 合伙人兑换码报名
export function wxPartnerRedemptionCode(activityID: number) {
  return http.post('/api/wx/Activities/partner-redemption', { activityID })
}

File: gin-vue-admin/server/router/wechat/activity.go
package wechat

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type WechatActivityRouter struct{}

func (s *WechatActivityRouter) InitWechatActivityRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	// 微信活动API不需要认证，放在PublicRouter中，使用api/wx前缀
	wechatActivityPublicRouter := PublicRouter.Group("api/wx")
	{
		wechatActivityPublicRouter.GET("Activities", wechatActivityApi.WxGetActivities)         // 微信小程序活动列表
		wechatActivityPublicRouter.GET("Activities/:id", wechatActivityApi.WxGetActivityDetail) // 微信小程序活动详情
	}

	// 需要认证的活动API路由
	wechatActivityRouter := Router.Group("api/wx").Use(middleware.JWTAuth())
	{
		wechatActivityRouter.POST("Activities/register", wechatActivityApi.WxRegisterForActivity)                     // 用户报名活动
		wechatActivityRouter.GET("Activities/registration-status", wechatActivityApi.WxGetActivityRegistrationStatus) // 获取活动报名状态
		wechatActivityRouter.POST("Activities/participant-info", wechatActivityApi.WxUpdateParticipantInfo)           // 更新参与者信息
		wechatActivityRouter.POST("Activities/partner-redemption", wechatActivityApi.WxPartnerRedemptionCode)         // 合伙人兑换码报名
	}
}
